src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -13,730 +13,874 @@ use rustc_mir::monomorphize;
 
 use traits::*;
 
+use std::borrow::Cow;
+
 use syntax::symbol::Symbol;
 use syntax_pos::Pos;
 
 use super::{FunctionCx, LocalRef};
 use super::place::PlaceRef;
-use super::operand::OperandRef;
+use super::operand::{OperandValue, OperandRef};
 use super::operand::OperandValue::{Pair, Ref, Immediate};
 
-impl<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
-    pub fn codegen_block(
-        &mut self,
-        bb: mir::BasicBlock,
-    ) {
-        let mut bx = self.build_block(bb);
-        let data = &self.mir[bb];
+/// Helper structure used by `FunctionCx::codegen_terminator`.
```

Comment:
Write something a bit more descriptive? E.g., ``Used by `FunctionCx::codegen_terminator` for ...`.